# HG changeset patch
# User Rik <rik@octave.org>
# Date 1471378453 25200
#      Tue Aug 16 13:14:13 2016 -0700
# Node ID 0e8c36397197f03d5398bbcfebd757bfece4298f
# Parent  6ca3acf5fad86d2d8335fcc171cd7879d026ca77
dump_demos.m: Replace 'x = y = XXX' and 'x = y = z = XXX' assignments.

* dump_demos.m: Replace 'x = y = XXX' and 'x = y = z = XXX' assignments.

diff --git a/scripts/testfun/private/dump_demos.m b/scripts/testfun/private/dump_demos.m
--- a/scripts/testfun/private/dump_demos.m
+++ b/scripts/testfun/private/dump_demos.m
@@ -217,16 +217,22 @@ function code = oct2mat (code)
   endfor
   ## Comment keywords unknown to Matlab
   commentkeywords = {"unwind_protect", "end_unwind_protect"};
   for k = 1:numel (commentkeywords)
     code = strrep (code, commentkeywords{k}, ["%" commentkeywords{k}]);
   endfor
   ## Fix in-place operators
   code = regexprep (code, '(\S+)(\s*)(.?[+-/*])=', '$1 = $1 $3');
+  ## Fix x = y = z = XXX assignment
+  code = regexprep (code, '^\s*([^=\s]+)\s*=\s*([^=\s]+)\s*=\s*([^=\s]+)\s*=\s*([^=\n]+)$', ...
+                          "$1 = $4\n$2 = $4\n$3 = $4", "lineanchors");
+  ## Fix x = y = XXX assignment
+  code = regexprep (code, '^\s*([^=\s]+)\s*=\s*([^=\s]+)\s*=\s*([^=\n]+)$', ...
+                          "$1 = $3\n$2 = $3", "lineanchors");
 
 endfunction
 
 function dump_helper_fcns (fid)
 
   fprintf (fid, "\n%s\n", repmat ("%", [1, 60]));
   fdisp (fid, "% Helper functions");
   fprintf (fid, "%s\n", repmat ("%", [1, 60]));
